src/admin/index.js

```diff
@@ -124,6 +124,52 @@ module.exports = ({
       }
     })
   }
+  /**
+   * @param {array} topicPartitions
+   * @param {boolean} [validateOnly=false]
+   * @param {number} [timeout=5000]
+   * @return {Promise}
+   */
+  const createPartitions = async ({ topicPartitions, validateOnly, timeout }) => {
+    if (!topicPartitions || !Array.isArray(topicPartitions)) {
+      throw new KafkaJSNonRetriableError(`Invalid topic partitions array ${topicPartitions}`)
+    }
+    if (topicPartitions.length === 0) {
+      throw new KafkaJSNonRetriableError(`Empty topic partitions array`)
+    }
+
+    if (topicPartitions.filter(({ topic }) => typeof topic !== 'string').length > 0) {
+      throw new KafkaJSNonRetriableError(
+        'Invalid topic partitions array, the topic names have to be a valid string'
+      )
+    }
+
+    const topicNames = new Set(topicPartitions.map(({ topic }) => topic))
+    if (topicNames.size < topicPartitions.length) {
+      throw new KafkaJSNonRetriableError(
+        'Invalid topic partitions array, it cannot have multiple entries for the same topic'
+      )
+    }
+
+    const retrier = createRetry(retry)
+
+    return retrier(async (bail, retryCount, retryTime) => {
+      try {
+        await cluster.refreshMetadata()
+        const broker = await cluster.findControllerBroker()
+        await broker.createPartitions({ topicPartitions, validateOnly, timeout })
+
+        return true
```

Comment:
Instead of resolving with `true`, let's just resolve with nothing. It's already clear that resolving means the operation worked and rejecting means it didn't. That leaves us room to add a proper response if future versions of the protocol starts returning some meaningful information that we need to expose.

Reply:
Done